### PR TITLE
NFTMania (Independent Publisher)

### PR DESCRIPTION
**Pull request by shreyan1999**
---
**Proposal for NFTMania Connector**
This is a proposed connector for _Microsoft Power Platform_. Blockchain is a tech space which is gaining its prominence and is expected to be a key player in the upcoming web 3.0, besides this is provides various features for the security of any digital asset. NFT is one such branch based on blockchain which is being used by many to store the collectibles like art or any piece of data digitally. The NFTs are Non Fungible in nature so the owner holds the right for any collectible. It cant be manipulated as the transaction data will be stored on Blockchain. Developing the NFT is often overlooked as a tedious task due to the coding involved. It takes a long time to get ideas to life due to this. The main purpose of this connector is to help developers bring their NFT applications to market in hours instead of months. This is possible with the help of Microsoft Power Platform.

**Detailed proposal in the readme.md file within the NFTMania Folder.**
**The Connector is also included in the Folder along with a readme file.**

### CODE REVIEW

**Comment by shreyan1999** [comment]
@Amjed-Ayoub This PR includes the proposal and the connector. Does this require a title change?

File: independent-publisher-connectors/NFTMania/MintNFT.swagger.json [hunk 1]
```diff
@@ -0,0 +1,84 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "MintNFT",
+    "description": "NFT Mania provides you with an Ultimate NFT experience and lets you develop applications in no time!",
+    "version": "1.0"
+  },
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/NFTMania/MintNFT.swagger.json:7]: Hi @shreyan1999, Please add a contact section, please see our templates [here](https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/Independent%20Publisher/apiDefinition.swagger.json)

File: independent-publisher-connectors/NFTMania/MintNFT.swagger.json [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  "basePath": "/",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [],
+  "produces": [],
+  "paths": {
+    "/v0/mints/easy/urls": {
+      "post": {
+        "responses": {
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/NFTMania/MintNFT.swagger.json:18]: @shreyan1999,, default responses are not allowed. Please define a proper response. please see our docs [here](https://learn.microsoft.com/en-us/connectors/custom-connectors/coding-standards#responses) and [here](https://learn.microsoft.com/en-us/connectors/custom-connectors/connector-string-guidance#operations)

File: independent-publisher-connectors/NFTMania/MintNFT.swagger.json [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+          "default": {
+            "description": "default",
+            "schema": {}
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/NFTMania/MintNFT.swagger.json:21]: @shreyan1999, empty Schema is not allowed, please define a proper schema or you can delete the empty schema.

File: independent-publisher-connectors/NFTMania/MintNFT.swagger.json [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+          }
+        },
+        "summary": "Easy minting with URL",
+        "description": "Easy minting with URL",
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/NFTMania/MintNFT.swagger.json:25]: @shreyan1999, all operations descriptions should end with a period, No period required only if there's a single word. Please see [here](https://learn.microsoft.com/en-us/connectors/custom-connectors/connector-string-guidance#operations)

File: independent-publisher-connectors/NFTMania/NFTMania.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
+**Proposal for NFTMania Connector**
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/NFTMania/NFTMania.md:1]: Please use this [template](https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/Independent%20Publisher/readme.md) for the readme file. you need to use the "#" key. you can see the raw code [here](https://raw.githubusercontent.com/microsoft/PowerPlatformConnectors/dev/templates/Independent%20Publisher/readme.md)

Please include the required sections.

File: independent-publisher-connectors/NFTMania/apiProperties.json [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+          "tooltip": "Provide your API Key",
+          "constraints": {
+            "tabIndex": 2,
+            "clearText": false,
+            "required": "true"
+          }
+        }
+      }
+    },
+    "iconBrandColor": "#007ee5",
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/NFTMania/apiProperties.json:18]: @shreyan1999, all independent publishers need to use "#da3b01" as their icon brand color

**Review by Amjed-Ayoub — CHANGES_REQUESTED** [review]

**Comment by Amjed-Ayoub** [comment]
Hello @shreyan1999,
I hope you are doing well.
I am following up with you to see how things are going with preparing your connector.
Please let us know if you have any questions.
Thank you.

**Comment by shreyan1999** [comment]
Hey Amjad! Working on the changes. Ill update here once they are ready.

File: independent-publisher-connectors/NFTMania/MintNFT.swagger.json (hunk 1 above)
Comment (shreyan1999) [reply independent-publisher-connectors/NFTMania/MintNFT.swagger.json:7]: Sorry for not providing it. I have added the contact section from the template.

**Comment by shreyan1999** [comment]
Hello @Amjed-Ayoub . Sorry for the delay. I have updated the files with all the required changes. Please let me know if I need to change anything else.

https://github.com/microsoft/PowerPlatformConnectors/pull/1948/commits/2f396506ea7d7c714dc28bd618beb4b63c34fd80

File: independent-publisher-connectors/NFTMania/apiDefinition.swagger.json [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+        "summary": "Mint NFT",
+        "description": "Mints NFT using the image URL",
+        "operationId": "Authorization",
+        "parameters": [
+          {
+            "name": "API",
+            "in": "header",
+            "required": true,
+            "type": "string",
+            "x-ms-visibility": "important"
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/NFTMania/apiDefinition.swagger.json:127]: @shreyan1999 The 'x-ms-summary' property is required, and the 'description' property is required. please add both.

File: independent-publisher-connectors/NFTMania/NFTMania.md (hunk 5 above)
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/NFTMania/NFTMania.md:1]: @shreyan1999 please follow this [template](https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/Independent%20Publisher/readme.md) and for the raw code view [here](https://raw.githubusercontent.com/microsoft/PowerPlatformConnectors/dev/templates/Independent%20Publisher/readme.md)

**Review by Amjed-Ayoub — CHANGES_REQUESTED** [review]

File: independent-publisher-connectors/NFTMania/NFTMania.md (hunk 5 above)
Comment (shreyan1999) [reply independent-publisher-connectors/NFTMania/NFTMania.md:1]: I have added it in the reade.md with the mentioned template.

Comment (Amjed-Ayoub) [inline independent-publisher-connectors/NFTMania/NFTMania.md:1]: @shreyan1999 please remove this file. Thank you

**Review by Amjed-Ayoub — DISMISSED** [review]

File: independent-publisher-connectors/NFTMania/NFTMania.md (hunk 5 above)
Comment (shreyan1999) [reply independent-publisher-connectors/NFTMania/NFTMania.md:1]: Okay. I will remove this file.

**Comment by shreyan1999** [comment]
Hello @Amjed-Ayoub I have removed the NFTMania.md file. Thanks a lot for the help and support! Looking forward to the next steps! I will push the update of this connector in the coming days. Thanks again!

**Comment by Amjed-Ayoub** [comment]
@shreyan1999 please remove the Proposal from the PR title.

**Comment by shreyan1999** [comment]
> @shreyan1999 please remove the Proposal from the PR title.

I have removed the Proposal from the title.

**Comment by Amjed-Ayoub** [comment]
[[certify-connector]]

**Comment by Amjed-Ayoub** [comment]
Hello @shreyan1999,
I hope you are doing well.
Congratulations, your pull request is approved. We will move forward with the certification process. We are preparing your connector for production deployment and will queue up for the next deployment schedule. Once your connector onboards the next deployment schedule, it starts to deploy your connector in our production environments that typically takes 3-4 weeks. 
Please let us know if you have any questions.
Thank you very much for working with us.

**Comment by Amjed-Ayoub** [comment]
Hello @shreyan1999,
I hope you are doing well.
Your connector is in production deployment and queued up for the next deployment schedule. This process typically takes 3-4 weeks. Please note that the documentation for your connector will not be generated until the deployment process is in the final steps.
Please let us know if you have any questions.
Thank you very much for working with us.

**Review by sriyen-msft — APPROVED** [review]

**Comment by shreyan1999** [comment]
Thanks a lot @Amjed-Ayoub for all the help during the process. 

Thank you @sriyen-msft . I have filled the feedback form.

**Pull request merged**